XDD Глобально доступный клиентский контекст для модульных тестов

petypen opened the pull request:
реализация  #818 
XDD Глобально доступный клиентский контекст для модульных тестов
сохранение и чтение глобального клиентского контекста

Реализована процедура, которая в браузере тестирования сохраняет пользовательский клиентский контекст. Значение контекста существует весь сеанс тестирования. Значение контекста хранится на клиенте и в отличии от контекста сохраненного методом `СохранитьКонтекст(Контекст)` не передается на сервер при выполнении внутренней логики браузера тестирования.

сделанные изменения

- Добавлен метод СохранитьГлобальныйКлиентскийКонтекст
- Добавлен метод ПолучитьГлобальныйКлиентскийКонтекст
- Добавлены тесты глобального клиентского контекста
- Добавлена документация глобального клиентского контекста


@vanessa-opensource/Collaborators - просьба прокомментировать и проверить

petypen committed:
feature: сохранение и чтение глобального клиентского
контекста

issues XDD Глобально доступный клиентский контекст для модульных тестов #818
-- Добавлен метод СохранитьГлобальныйКлиентскийКонтекст
-- Добавлен метод ПолучитьГлобальныйКлиентскийКонтекст

petypen committed:
tests: тесты глобального клиентского контекста

petypen committed:
doc: документация глобального клиентского контекста

artbear commented:
@petypen Интересная тема.

предлагаю наименование методов упростить до
- СохранитьГлобальныйКонтекст
- ПолучитьГлобальныйКонтекст
без использования Клиентский

artbear commented:
По опыту использования контекстов удобно не задавать их каждый раз через указанные методы, а сразу задавать коллекцию ключей-задачей в виде структуры (предпочтительно) или соответствия 

и использовать АПИ для управления или даже без АПИ, всегда зная, что такой Контекст - это структура

и юзать следующее АПИ контекста
- ГлобальныйКонтекст()
- ЗначениеИзГлобальногоКонтекста(Ключ)
- ДобавитьВГлобальныйКонтекст(Ключ, Значение)
- УдалитьИзГлобальногоКонтекста(Ключ)

petypen committed:
feature: уточнения по рекомендациям @artbear

-- Глобальный контекст теперь структура
-- https://github.com/vanessa-opensource/add/pull/820#issuecomment-745485931

petypen committed:
doc: уточнена документация

petypen committed:
tests: Уточнены тесты

petypen commented:
@artbear Артур, добрый день. Спасибо за дельные замечания.
Сделал PR с учетом предложенного API контекста.

-- доработан браузер тестирования
-- уточнена документация
-- уточнены тесты

artbear commented on epf/xddTestRunner/xddTestRunner/Forms/УправляемаяФорма/Ext/Form/Module.bsl:
```diff
@@ -2755,3 +2762,106 @@
+	
+КонецПроцедуры
+
+#КонецОбласти
```
на последней строке лучше сделать перенос строки, тогда гит будет учитывать историю этой последней строки.

artbear commented on epf/xddTestRunner/xddTestRunner/Forms/УправляемаяФорма/Ext/Form/Module.bsl:
```diff
@@ -2755,3 +2762,106 @@
+Функция ГлобальныйКонтекстЗначение(Знач Ключ) Экспорт
+	Перем ЗначениеКонтекста;
+	
+	Если ТипЗнч(ГлобальныйКонтекст) <> Тип("Структура") Тогда
```
Предлагаю этот блок удалить

ведь глоб.контекст в любом случае уже должен быть инициализирован в новую структуру перед запуском тестов.

иначе наш тест не будет честным.

```bsl
	Если ТипЗнч(ГлобальныйКонтекст) <> Тип("Структура") Тогда
		ГлобальныйКонтекстИнициализировать();
	КонецЕсли;
```

artbear replied:
Проверь, плиз, тест после этой правки.

может быть, первичная инициализация не в том месте выполняется?

artbear commented on epf/xddTestRunner/xddTestRunner/Forms/УправляемаяФорма/Ext/Form/Module.bsl:
```diff
@@ -2755,3 +2762,106 @@
+&НаКлиенте
+Процедура ГлобальныйКонтекстДобавить(Знач Ключ, Знач ЗначениеКонтекста) Экспорт
+	
+	Если ТипЗнч(ГлобальныйКонтекст) <> Тип("Структура") Тогда
```
и здесь удалить

artbear commented on epf/xddTestRunner/xddTestRunner/Forms/УправляемаяФорма/Ext/Form/Module.bsl:
```diff
@@ -2755,3 +2762,106 @@
+&НаКлиенте
+Процедура ГлобальныйКонтекстУдалить(Знач Ключ) Экспорт
+	
+	Если ТипЗнч(ГлобальныйКонтекст) <> Тип("Структура") Тогда
```
и здесь удалить

artbear reviewed (changes requested):
предлагаю немного доработать.

petypen committed:
refactor: учтены замечания @artbear

https://github.com/vanessa-opensource/add/pull/820#pullrequestreview-565491554

-- Удалена излишняя инициализация глобального контекста в методах:
* ГлобальныйКонтекстЗначение
* ГлобальныйКонтекстДобавить
* ГлобальныйКонтекстУдалить

petypen commented:
@artbear  Артур, учел замечания.

artbear reviewed (approved):
Отлично. @petypen Большое спасибо за терпение!
Делаю мерж.
